pkg/operation/seed/seed.go

```diff
@@ -828,151 +828,351 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
+	}
 
-// DebootstrapCluster deletes certain resources from the seed cluster.
-func DebootstrapCluster(ctx context.Context, k8sSeedClient kubernetes.Interface) error {
-	bootstrapComponents, err := bootstrapComponents(k8sSeedClient, v1beta1constants.GardenNamespace, nil, nil)
+	grmImage, err := imageVector.FindImage(charts.ImageNameGardenerResourceManager, imagevector.RuntimeVersion(sc.Version()), imagevector.TargetVersion(sc.Version()))
 	if err != nil {
 		return err
 	}
-
-	// Delete component specific resources
-	var debootstrapFunctions []flow.TaskFn
-	for _, componentFn := range bootstrapComponents {
-		fn := componentFn
-		debootstrapFunctions = append(debootstrapFunctions, func(ctx context.Context) error {
-			return component.OpDestroyAndWait(fn).Destroy(ctx)
-		})
+	cfg := resourcemanager.Values{
+		ConcurrentSyncs:  pointer.Int32Ptr(20),
+		HealthSyncPeriod: utils.DurationPtr(time.Minute),
+		ResourceClass:    pointer.StringPtr(v1beta1constants.SeedResourceManagerClass),
+		SyncPeriod:       utils.DurationPtr(time.Hour),
 	}
+	rm := resourcemanager.New(sc.Client(), namespace, grmImage.String(), 1, cfg)
 
-	if err = flow.Parallel(debootstrapFunctions...)(ctx); err != nil {
+	etcdImage, err := imageVector.FindImage(charts.ImageNameEtcdDruid, imagevector.RuntimeVersion(sc.Version()), imagevector.TargetVersion(sc.Version()))
+	if err != nil {
 		return err
 	}
+	var etcdImageVectorOverwrite *string
+	if val, ok := imageVectorOverwrites[etcd.Druid]; ok {
+		etcdImageVectorOverwrite = &val
+	}
+	etcdDruid := etcd.NewBootstrapper(sc.Client(), namespace, etcdImage.String(), kubernetesVersion, etcdImageVectorOverwrite)
 
-	return destroyGardenerResourceManager(ctx, k8sSeedClient)
-}
-
-func destroyGardenerResourceManager(ctx context.Context, c kubernetes.Interface) error {
-	managedResources := &resourcesv1alpha1.ManagedResourceList{}
-	if err := c.Client().List(ctx, managedResources, client.InNamespace(v1beta1constants.GardenNamespace)); err != nil {
+	gardenerSeedAdmissionControllerImage, err := imageVector.FindImage(charts.ImageNameGardenerSeedAdmissionController)
+	if err != nil {
 		return err
 	}
-	if len(managedResources.Items) == 0 {
-		rm := resourcemanager.New(c.Client(), v1beta1constants.GardenNamespace, "", 0, resourcemanager.Values{})
-		return rm.Destroy(ctx)
+	var (
+		repository = gardenerSeedAdmissionControllerImage.String()
+		tag        = version.Get().GitVersion
+	)
+	if gardenerSeedAdmissionControllerImage.Tag != nil {
+		repository = gardenerSeedAdmissionControllerImage.Repository
+		tag = *gardenerSeedAdmissionControllerImage.Tag
 	}
-	return errors.New("cannot delete gardener-resource-manager there are managed resources in the garden namespace left")
-}
+	gsacImage := imagevector.Image{
+		Repository: repository,
+		Tag:        &tag,
+	}
+	seedAdmission := seedadmissioncontroller.New(sc.Client(), namespace, gsacImage.String(), kubernetesVersion)
```

Comment:
You use three names to describe the "same" thing: `gardenerSeedAdmissionController`, `gsac`, `seedAdmission`. I suggest to streamline this.